### PR TITLE
Functions to calculate the closest bigger city than the given one and up to the biggest in the country.

### PR DESCRIPTION
**Pull request by kermit10000000**
The script works and you have a given example in main.cpp
Maybe you have some idea of further usage of this like for each capital choosen by the best capital scritpt or something else? :) 

Signed-off-by: Wojciech Mielczarek <kermit10000000@gmail.com>

### CODE REVIEW

File: ClosestCity/ClosestCity.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ */
+
+#include "ClosestCity.hpp"
+
+namespace ClosestCity {
+
+void removeCites(std::vector<City> &vec, const std::string &cityName) {
```
Comment (JaroslawWiosna) [inline ClosestCity/ClosestCity.cpp]: pls, rename to `removeSmallerCitiesInVector` or `removeSmallerCities`

File: main.cpp [hunk 2]
```diff
@@ -59,12 +60,16 @@ int main(int argc, char *argv[]) {
 
     VectorOfCities::setDistanceToTheFarthestInTheWholeVector(vec);
 
+
```
Comment (JaroslawWiosna) [inline main.cpp]: remove this line

File: ClosestCity/ClosestCity.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    } else {
+        return " ";
+    }
+}
+std::string getAllBiggestCitiesInOrder(std::vector<City> vec,
+                                       std::string cityName) {
+    std::string cities_list = cityName;
+    std::string next_city_in_line = "";
+    while (cityName != " ") {
+        removeCites(vec, cityName);
```
Comment (JaroslawWiosna) [inline ClosestCity/ClosestCity.cpp]: Please think about removing this line.
Perhaps it is not needed for performance.... But for now, you can merge it.

**Review by JaroslawWiosna — APPROVED** [review]

**Comment by kermit10000000** [comment]
1. fixed
2. performance to be checked in the future
3. fixed
4. empty() instead of  " " fails for some unknown reason. maybe changed in the future.

**Comment by kermit10000000** [comment]
I know this pull request is closed but since we discussed it here, about second point 3 options should be checked: 1. performance with no remove function, 2. performance with remove in while(!="") and 3 . only once called remove function before the while(!="")

**Pull request merged**